tests/HookFilter.ts

```diff
@@ -3,29 +3,56 @@ import { HookFilter } from '../src/HookFilter';
 
 const DEFAULT_SET = ['default-allowed', 'default-allowed-but-ignored'];
 const ENABLED_SET = ['enabled-hook', 'enabled-but-ignored'];
-const IGNORED_SET = ['ignored', 'enabled-but-ignored', 'default-allowed-but-ignored'];
 
 describe("HookFilter", () => {
     let filter: HookFilter<string>;
     beforeEach(() => {
-        filter = new HookFilter(DEFAULT_SET, ENABLED_SET, IGNORED_SET);
+        filter = new HookFilter(ENABLED_SET);
     });
-    it('should skip a hook named in ignoreHooks', () => {
-        expect(filter.shouldSkip('ignored')).to.be.true;
+    describe('shouldSkip', () => {
+        it('should allow a hook named in enabled set', () => {
+            expect(filter.shouldSkip('enabled-hook')).to.be.false;
+        });
+        it('should not allow a hook not named in enabled set', () => {
+            expect(filter.shouldSkip('not-enabled-hook')).to.be.true;
+        });
     });
-    it('should allow a hook named in defaults', () => {
-        expect(filter.shouldSkip('default-allowed')).to.be.false;
-    });
-    it('should allow a hook named in enabled', () => {
-        expect(filter.shouldSkip('enabled-hook')).to.be.false;
-    });
-    it('should skip a hook named in defaults but also in ignored', () => {
-        expect(filter.shouldSkip('default-allowed-but-ignored')).to.be.true;
-    });
-    it('should skip a hook named in enabled but also in ignored', () => {
-        expect(filter.shouldSkip('enabled-but-ignored')).to.be.true;
-    });
-    it('should skip if any hooks are in ignored', () => {
-        expect(filter.shouldSkip('enabled-hook', 'enabled-but-ignored')).to.be.true;
+    
+    describe('convertIgnoredHooksToEnabledHooks', () => {
+        it('should correctly provide a list of default hooks', () => {
+            expect(HookFilter.convertIgnoredHooksToEnabledHooks([], [], DEFAULT_SET)).to.deep.equal(
+                DEFAULT_SET
+            );
+        });
+        
+        it('should correctly include both default and enabled hooks', () => {
```

Comment:
This should only return the enabled set since defaults only come into play with ignored hooks.

Reply:
Our assumptions were wrong in review. `convertIgnoredHooksToEnabledHooks` is only called when ignoreHooks is non-falsey, so we can assume that it will always have values. The test here is fine.